src/SignalR/clients/ts/signalr/src/Utils.ts

```diff
@@ -181,3 +187,72 @@ export class ConsoleLogger implements ILogger {
         }
     }
 }
+
+/** @private */
+export function getUserAgentHeader(): [string, string] {
+    let userAgentHeaderName = "X-SignalR-UserAgent";
+    if (Platform.isNode) {
+        userAgentHeaderName = "User-Agent";
+    }
+    return [ userAgentHeaderName, constructUserAgent(VERSION, getOsName(), getRuntime(), getRuntimeVersion()) ];
+}
+
+/** @private */
+export function constructUserAgent(version: string, os: string, runtime: string, runtimeVersion: string | undefined): string {
+    // Microsoft SignalR/[Version] ([Detailed Version]; [Operating System]; [Runtime]; [Runtime Version])
+    let userAgent: string = "Microsoft SignalR/";
+
+    const majorAndMinor = version.split(".");
+    userAgent += `${majorAndMinor[0]}.${majorAndMinor[1]}`;
+    userAgent += ` (${version}; `;
+
+    if (os && os !== "") {
+        userAgent += `${os}; `;
+    } else {
+        userAgent += "Unknown OS; ";
+    }
+
+    userAgent += `${runtime}`;
+
+    if (runtimeVersion) {
+        userAgent += `; ${runtimeVersion}`;
+    } else {
+        userAgent += "; Unknown Runtime Version";
+    }
+
+    userAgent += ")";
+    return userAgent;
+}
+
+function getOsName(): string {
+    let os: string = "";
+    if (Platform.isNode) {
+        os = process.platform;
```

Comment:
nit: can't you just put the switch inside of this if?

Reply:
Also after a quick google search, there seem to be ways to know the os platform from pure javascript. Seems super hacky though: https://stackoverflow.com/questions/9514179/how-to-find-the-operating-system-version-using-javascript

Reply:
Yeah, not doing that :)